### ISSUE TITLE
Latest Brixton snapshot does not allow Config First Bootstrapping

### ISSUE THREAD

**Issue by TabberGit** [spring-cloud/spring-cloud-config#322]
Here are the BOMs I'm running with:

     spring_boot_version:1.3.2.BUILD-SNAPSHOT
     spring_cloud_version:Brixton.BUILD-SNAPSHOT
     spring_cloud_config_version:1.1.0.BUILD-SNAPSHOT

I pulled code this afternoon, and I can longer run a simple config server in Config First Bootstrapping mode.  Instead, it goes to Jersey and attempts to contact a potential discovery server at localhost:8761.  I have even tried `spring.cloud.config.discovery.enabled=false` to no avail.

**Comment by dsyer** [comment]
This sample still works: https://github.com/spring-cloud-samples/tests/tree/master/configserver-bootstrap. Do you have some code you can share?

**Comment by TabberGit** [comment]
Sure, the main app on this is extremely simple

    @EnableAutoConfiguration
    @EnableConfigServer
    @ComponentScan
    @Slf4j
    @RestController
    public class Application {
        @RequestMapping("/home")
        public String home() {
            log.debug("configserver home called");
            return "This is the configserver";
        }

        public static void main(String[] args) {
            SpringApplication.run(Application.class, args);
        }
    }

**bootstrap.yml**

    spring:
      application:
        name: ${APPLICATION_NAME}
    

**application.yml**

    info:
      component: Config Server
    
    server:
      port: 8888
    
    management:
      context-path: /admin
    
    spring:
      cloud:
        config:
          server:
            git:
              uri: ssh://git@stash.csgsystems.com:7999/~bout01/config-repo.git
              basedir: config-repo
        stream:
           bindings:
             springCloudBusInput:
               group: ${APPLICATION_NAME}.${spring.cloud.client.hostname}.${server.port}
      profiles:
        active: development

**Comment by dsyer** [comment]
You can't bootstrap the config location from `application.yml`. Compare your app with the sample I said was working and you will see what I mean.

**Comment by TabberGit** [comment]
Very curious that we were getting away with doing it this way since at least Nov. 16th of last year.  I'll make the changes you suggested & let you know what I find.

**Comment by TabberGit** [comment]
So...  It seems that having the eureka client in the classpath has become an issue.  It was being carried in by a homegrown starter that we made.  Once that was removed, it stopped trying to register.

However, even with eureka in the classpath, it is strange that it would automatically engage it without any annotations instructing it to do so (beyond this issue thread.)

I tried the config location in both the bootstrap.yml and application.yml files once the eureka jars were removed with success.  I can't think of a reason why we _need_ to have the config location in application.yml.  I'm just uncertain as to why it can't be placed there.
